src/python/pants/option/options_bootstrapper.py

```diff
@@ -213,26 +213,25 @@ def verify_configs_against_options(self, options: Options) -> None:
     error_log = []
     for config in self.config.configs():
       for section in config.sections():
-        if section == GLOBAL_SCOPE_CONFIG_SECTION:
-          scope = GLOBAL_SCOPE
-        else:
-          scope = section
+        scope = GLOBAL_SCOPE if section == GLOBAL_SCOPE_CONFIG_SECTION else section
         try:
           valid_options_under_scope = set(options.for_scope(scope))
         # Only catch ConfigValidationError. Other exceptions will be raised directly.
         except Config.ConfigValidationError:
-          error_log.append(f"Invalid scope [{section}] in {config.configpath}")
+          error_log.append(f"Invalid scope [{section}] in {config.config_path}")
         else:
           # All the options specified under [`section`] in `config` excluding bootstrap defaults.
-          all_options_under_scope = (set(config.configparser.options(section)) -
-                                     set(config.configparser.defaults()))
```

Comment:
This is the leaky abstraction referred to in the PR description. It's the only place in the options code where we are directly interfacing with a `configparser.ConfigParser`, whereas everything else goes through our `Config` encapsulation.

Reply:
Could we add a `TODO` here then?

Reply:
Nothing to `TODO` anymore - this PR fixes the leaky abstraction! Now a `_TomlValues` or `_YamlValues` class in `config.py` will need to define `options()` and `defaults()` methods, which means that this code in `options_bootstrapper.py` will continue to work without any change 🎉